grpc/src/test/java/com/linecorp/armeria/client/grpc/GrpcClientAdditionalAuthorityTest.java

```diff
@@ -0,0 +1,222 @@
+            }
+        };
+
+        final TestServiceBlockingStub client = GrpcClients.builder(protocol + "://foo.com:" + server.httpPort())
+                                                          .factory(clientFactory)
+                                                          .setHeader(headerName, "baz.com")
+                                                          .intercept(authorityOverridingInterceptor)
+                                                          .build(TestServiceBlockingStub.class);
+
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            final SimpleResponse response = client.unaryCall(SimpleRequest.getDefaultInstance());
+            assertThat(response.getUsername()).isEqualTo("BarService");
+            assertThat(captor.get().endpoint().authority())
+                    .isEqualTo("foo.com:" + server.httpPort());
+        }
+    }
+
+    @CsvSource({ "h1c", "h2c", "http", })
+    @ParameterizedTest
+    void shouldUseEndpointAsAuthority(String protocol) {
+        final TestServiceBlockingStub client = GrpcClients.builder(protocol + "://foo.com:" + server.httpPort())
+                                                          .factory(clientFactory)
+                                                          .build(TestServiceBlockingStub.class);
+
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            final SimpleResponse response = client.unaryCall(SimpleRequest.getDefaultInstance());
+            assertThat(response.getUsername()).isEqualTo("FooService");
+            assertThat(captor.get().endpoint().authority())
+                    .isEqualTo("foo.com:" + server.httpPort());
+        }
+    }
+
+    @CsvSource({ "h1c, :authority", "h2c, :authority", "http, :authority",
+                 "h1c, Host", "h2c, Host", "http, Host" })
+    @ParameterizedTest
+    void shouldOverrideCallAuthorityWithAdditionalAuthority(String protocol, String headerName) {
+        final ClientInterceptor authorityOverridingInterceptor = new ClientInterceptor() {
+            @Override
+            public <I, O> ClientCall<I, O> interceptCall(
+                    MethodDescriptor<I, O> method,
+                    CallOptions callOptions, Channel next) {
+                return next.newCall(method, callOptions.withAuthority("bar.com"));
+            }
+        };
+
+        final TestServiceBlockingStub client = GrpcClients.builder(protocol + "://foo.com:" + server.httpPort())
+                                                          .factory(clientFactory)
+                                                          .intercept(authorityOverridingInterceptor)
+                                                          .build(TestServiceBlockingStub.class);
+
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(headerName, "bar.com"));
```

Comment:
Could you use different value from the interceptor so that we can be sure which one is applied?